test/gtest/uct/cuda/test_cuda_ipc_md.cc

```diff
@@ -41,6 +41,60 @@ class test_cuda_ipc_md : public test_md {
         EXPECT_EQ(CUDA_SUCCESS, cuMemFree(ptr));
         return rkey;
     }
+
+#if HAVE_CUDA_FABRIC
+    static uct_cuda_ipc_rkey_t unpack_masync(uct_md_h md, int64_t uuid)
+    {
+        size_t size = (1 << 22);
+        CUmemPoolProps pool_props = {};
+        CUdeviceptr ptr;
+        CUmemoryPool mpool;
+        CUmemAccessDesc map_desc;
+        CUstream cu_stream;
+        CUdevice cu_device;
+
+        EXPECT_EQ(CUDA_SUCCESS, cuCtxGetDevice(&cu_device));
+
+        pool_props.allocType     = CU_MEM_ALLOCATION_TYPE_PINNED;
+        pool_props.location.id   = (int)cu_device;
+        pool_props.location.type = CU_MEM_LOCATION_TYPE_DEVICE;
+        pool_props.handleTypes   = CU_MEM_HANDLE_TYPE_FABRIC;
+        pool_props.maxSize       = size;
+        map_desc.flags           = CU_MEM_ACCESS_FLAGS_PROT_READWRITE;
+        map_desc.location        = pool_props.location;
+
+        EXPECT_EQ(CUDA_SUCCESS, cuStreamCreate(&cu_stream, CU_STREAM_NON_BLOCKING));
+        EXPECT_EQ(CUDA_SUCCESS, cuMemPoolCreate(&mpool, &pool_props));
+        EXPECT_EQ(CUDA_SUCCESS, cuMemPoolSetAccess(mpool, &map_desc, 1));
+        EXPECT_EQ(CUDA_SUCCESS, cuMemAllocFromPoolAsync(&ptr, size, mpool, cu_stream));
+        EXPECT_EQ(CUDA_SUCCESS, cuStreamSynchronize(cu_stream));
+
+        uct_mem_h memh;
```

Comment:
The part below is identical to existing `unpack` method, except the allocation/free part.
Maybe for code reuse purpose we encapsulate the common part into `unpack_impl` method, that does only reg-pack-unpack-dereg without memory alloc/free?